src/server.ts

```diff
@@ -121,11 +129,30 @@ export class Server extends EventEmitter {
    * Two different clients SHOULD NOT be given the same address and secret.
    *
    * @param connectionTag Optional connection identifier that will be appended to the ILP address and can be used to identify incoming connections. Can only include characters that can go into an ILP Address
+   * @param receiptNonce Optional nonce to include in STREAM receipts
+   * @param receiptSecret Optional secret to use for signing STREAM receipts
    */
-  generateAddressAndSecret (connectionTag?: string): { destinationAccount: string, sharedSecret: Buffer } {
+  generateAddressAndSecret (opts?: string | GenerateAddressSecretOpts): { destinationAccount: string, sharedSecret: Buffer } {
```

Comment:
If the receipt nonce and secret are appended as a separate segment, it'd be easier for intermediaries to identify, remove or alter it (is that a concern?). Also, any logic elsewhere that only uses the last segment of the ILP address to identify a connection may work incorrectly after this change

What if the connection ID, tag, receipt nonce and secret were all encoded into a single encrypted blob/token so they can't be separately tracked or modified?

Reply:
I actually had the nonce and secret encoded with the tag initially (https://github.com/interledgerjs/ilp-protocol-stream/pull/133/commits/82981060a513138d7cc2d4c76fc714f24d480a4f) but changed it (https://github.com/interledgerjs/ilp-protocol-stream/pull/133/commits/757ccff5ce39858abbc0541092caba6fdb18bbad).

I think I do like the idea of making them part of the connection token (like the connection tag is currently) in order to make tampering detectable.

Reply:
The connection tag and receipt nonce are still opaque/can be tampered with or removed. Instead of only encrypting the receipt secret using the server secret -- what if the whole last segment of the address (id, tag, receipt nonce, receipt secret) was encrypted using the server secret?

Reply:
I think they're tamper-proof since they're part of the id/token used to generate the shared secret and verified at the start of the connection: 
https://github.com/interledgerjs/ilp-protocol-stream/blob/d2c6b14fd2ff7266b37adbc46c45834f2d55de6f/src/pool.ts#L93-L109
If that's correct, then I was thinking still just encrypt the receipt secret to keep the destination address from being even loooonger.

Reply:
Ah, gotcha -- yeah, that should be tamper proof.

> If that's correct, then I was thinking still just encrypt the receipt secret to keep the destination address from being even loooonger.

How much longer would it be if it was encrypted?

One idea to shorten it: could we use the receipt nonce as the ID for the connection instead of randomly generating another token?

Reply:
Actually it looks like the encryption used always adds an extra 28 bytes. (I was previously scared off by the fact that the receipt secret size went from 32 bytes to 60 bytes encrypted.)
So I can encode and encrypt the token, connection tag, receipt nonce, and receipt secret all together.

I think the STREAM server should still generate its own ID instead of relying on unique receipt nonces in the (proxied) SPSP query.